vector/src/main/java/im/vector/app/features/home/room/list/RoomListSectionBuilderSpace.kt

```diff
@@ -429,27 +429,24 @@ class RoomListSectionBuilderSpace(
     }
 
     private fun withQueryParams(builder: (RoomSummaryQueryParams.Builder) -> Unit, block: (RoomSummaryQueryParams) -> Unit) {
-        RoomSummaryQueryParams.Builder()
-                .apply { builder.invoke(this) }
-                .build()
-                .let { block(it) }
+        block(roomSummaryQueryParams { builder.invoke(this) })
     }
 
     internal fun RoomSummaryQueryParams.process(spaceFilter: RoomListViewModel.SpaceFilterStrategy, currentSpace: String?): RoomSummaryQueryParams {
         return when (spaceFilter) {
             RoomListViewModel.SpaceFilterStrategy.ORPHANS_IF_SPACE_NULL -> {
                 copy(
-                        activeSpaceFilter = ActiveSpaceFilter.ActiveSpace(currentSpace)
+                        spaceFilter = currentSpace?.let { SpaceFilter.ActiveSpace(it) } ?: SpaceFilter.OrphanRooms
                 )
             }
             RoomListViewModel.SpaceFilterStrategy.ALL_IF_SPACE_NULL     -> {
                 if (currentSpace == null) {
                     copy(
-                            activeSpaceFilter = ActiveSpaceFilter.None
+                            spaceFilter = null
                     )
                 } else {
                     copy(
-                            activeSpaceFilter = ActiveSpaceFilter.ActiveSpace(currentSpace)
+                            spaceFilter = SpaceFilter.ActiveSpace(currentSpace)
```

Comment:
I was tempted to rewrite to something like
`spaceFilter = currentSpace?.let{ SpaceFilter.ActiveSpace(it) }`
But I think this is better to keep it like that for code clarity.

Reply:
Imo it's still very clear and it would be nice to be consistent with the change above in line 439. There would be a slight difference in functionality though. What happens if we get `SpaceFilter.ActiveSpace(null)`?

Reply:
Using null in `SpaceFilter.ActiveSpace(null)` is not possible anymore.

Reply:
f5194d30325c844ce523c46b62eaea58c17b396b